freeipa/src/main/java/com/sequenceiq/freeipa/converter/stack/StackToCreateFreeIpaRequestConverter.java

```diff
@@ -0,0 +1,409 @@
+        return request;
+    }
+
+    private BackupRequest getBackup(Stack stack) {
+        BackupRequest request = null;
+        Backup backup = stack.getBackup();
+        if (backup != null) {
+            request = new BackupRequest();
+            request.setStorageLocation(getBackupLocation(stack, backup.getStorageLocation()));
+            request.setS3(backup.getS3());
+            request.setAdlsGen2(backup.getAdlsGen2());
+            request.setGcs(backup.getGcs());
+        }
+        LOGGER.debug("Created backup request {} from backup {}", request, backup);
+        return request;
+    }
+
+    private Map<String, String> getTags(Json tags) {
+        Map<String, String> userDefined = Maps.newHashMap();
+        if (tags != null) {
+            try {
+                StackTags stackTag = tags.get(StackTags.class);
+                userDefined.putAll(stackTag.getUserDefinedTags());
+            } catch (IOException e) {
+                LOGGER.info("Exception during converting user defined tags.", e);
+            }
+        }
+        LOGGER.debug("Created tags request {} from tags {}", userDefined, tags);
+        return userDefined;
+    }
+
+    private String getLoggingLocation(String location) {
+        return getLocation(location, CLUSTER_LOG_PREFIX);
+    }
+
+    String getBackupLocation(Stack stack, String location) {
+        String originalLocation = getLocation(location, CLUSTER_BACKUP_PREFIX);
+
+        // During provisioning, the URL for Azure is converted from abfs/abfss to https, this conversion needs to be reversed.
+        // Example: convert https://storage1.dfs.core.windows.net/logs-fs into abfs://logs-fs@storage1.dfs.core.windows.net
+        try {
+            if (CloudPlatform.AZURE.equalsIgnoreCase(stack.getCloudPlatform()) && originalLocation != null) {
+                URI uri = new URI(originalLocation);
+                if (AZURE_BLOB_STORAGE_SCHEMA.equals(uri.getScheme())) {
+                    String uriPath = uri.getPath();
+                    int firstSeparator = uriPath.indexOf(PATH_DELIMETER);
+                    if (firstSeparator != -1) {
+                        int secondSeparator = uriPath.indexOf(PATH_DELIMETER, firstSeparator + 1);
+                        String bucketName;
+                        String bucketPath = "";
+                        if (secondSeparator == -1) {
+                            bucketName = uriPath.substring(firstSeparator + 1);
+                        } else {
+                            bucketName = uriPath.substring(firstSeparator + 1, secondSeparator);
+                            bucketPath = uriPath.substring(secondSeparator);
+                        }
+                        originalLocation = String.format("%s%s@%s%s", ORIGINAL_AZURE_BLOB_STORAGE_SCHEMA, bucketName, uri.getHost(), bucketPath);
+                    }
+                }
+            }
+        } catch (URISyntaxException e) {
+            LOGGER.error("URI syntax error");
```

Comment:
are you sure we should ignore this error and move on with the rebuild?

Reply:
I changed it to throw an exception.